### PR TITLE
Added data column generation to the data pipe, pagination as last pipe, Fix dts files 

### PR DESCRIPTION
**Pull request by klumba12**

### CODE REVIEW

File: src/core/infrastructure/event.d.ts [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-  (): void;
+	(): void;
 }
 
 export declare class Event {
-  constructor(e?: () => any);
+	constructor(e?: () => any);
 
-  on(f: IFunc): IReturn;
+	on(handler: IFunc): IReturn;
```
Comment (titovmx) [inline src/core/infrastructure/event.d.ts:10]: maybe rename ```IFunc``` to something like ```IEventHandler```?

File: src/core/infrastructure/event.d.ts [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 
-  watch(f: IFunc): void;
+	watch(handler: (e: any, off?: () => void) => void): () => void;
```
Comment (titovmx) [inline src/core/infrastructure/event.d.ts:12]: really hard to read )

Comment (titovmx) [reply src/core/infrastructure/event.d.ts:12]: what's difference between this and ```IFunc```?

Comment (klumba12) [reply src/core/infrastructure/event.d.ts:12]: this is Func of Func)

Comment (klumba12) [reply src/core/infrastructure/event.d.ts:12]: more complicated then just IFunc

Comment (klumba12) [reply src/core/infrastructure/event.d.ts:12]: will make better for ng2 later

**Review by titovmx — APPROVED** [review]

**Pull request merged**